src/wovnio/wovnphp/custom_domain/CustomDomainLangs.php

```diff
@@ -6,14 +6,37 @@
 class CustomDomainLangs
 {
     private $customDomainLangs;
-    public function __construct($customDomainLangsSettingsArray)
+
+
+    public function __construct($customDomainLangsSettingsArray, $defaultLang)
     {
+        $defaultLangCustomDomain = CustomDomainLangs::getDefaultLangCustomDomain($customDomainLangsSettingsArray, $defaultLang);
         $this->customDomainLangs = array();
-        foreach ($customDomainLangsSettingsArray as $langUrl => $lang) {
+        foreach ($customDomainLangsSettingsArray as $langUrl => $config) {
             $parsedUrl = parse_url($this->addProtocolIfNeeded($langUrl));
+            if (is_array($config)) {
+                $source = array_key_exists('source', $config) ? $config['source'] : $defaultLangCustomDomain;
+            } else {
+                $config = array('lang' => $config);
+                $source = $defaultLangCustomDomain;
+            }
 
             // Disable notice error by adding @, when path is not defined
-            array_push($this->customDomainLangs, new CustomDomainLang($parsedUrl['host'], @$parsedUrl['path'], $lang));
+            array_push($this->customDomainLangs, new CustomDomainLang($parsedUrl['host'], @$parsedUrl['path'], $config['lang'], $source));
+        }
+    }
+
+    public function getSourceCustomDomainByLang($langCode)
+    {
+        $results = array_filter($this->customDomainLangs, function ($customDomain) use ($langCode) {
```

Comment:
This function looks very similar to `getCustomDomainLangByLang`. I suspect you could be using it internally.